### PR TITLE
chore: refactoring sync and async servers

### PR DESCRIPTION
**Pull request by ashwinidulams**
changes related to refactoring of
sync for map, mapt
async for map, mapt and reduce.

### CODE REVIEW

File: pynumaflow/function/async_server.py [hunk 1]
```diff
@@ -55,59 +52,7 @@ async def datum_generator(
...
-    ...
-    >>> grpc_server = UserDefinedFunctionServicer(
-    ...   reduce_handler=reduce_handler,
-    ...   mapt_handler=mapt_handler,
-    ...   map_handler=map_handler,
-    ... )
-    >>> aiorun.run(grpc_server.start_async())
-    """
-
+class AsyncServer(udfunction_pb2_grpc.UserDefinedFunctionServicer):
```
Comment (ab93) [inline pynumaflow/function/async_server.py:55]: Shall we add docstrings for this one?

File: pynumaflow/function/async_server.py [hunk 2]
```diff
@@ -146,74 +91,72 @@ def MapFn(
...
-            datums.append(
-                udfunction_pb2.Datum(
-                    keys=list(msgt.keys),
-                    value=msgt.value,
-                    event_time=udfunction_pb2.EventTime(event_time=event_time_timestamp),
-                    watermark=udfunction_pb2.Watermark(watermark=watermark_timestamp),
-                )
-            )
-        return udfunction_pb2.DatumList(elements=datums)
+    # def MapTFn(
```
Comment (ab93) [inline pynumaflow/function/async_server.py:121]: do we need to keep this commented block?

File: pynumaflow/function/multiproc_server.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+
+class MultiProcServer(udfunction_pb2_grpc.UserDefinedFunctionServicer):
+    """
+    MultiProcServer Class a multiprocessing implementation of the grpc server.
+    They use a given TCP socket with SOCK_REUSE to allow all servers,
+    to listen on the same port. We start servers equal to the CPU count of the
+    system.
+
+    Arguments
```
Comment (ab93) [inline pynumaflow/function/multiproc_server.py:48]: I think we use `Args` at other places. Let's just be consistent. :)

File: pynumaflow/function/multiproc_server.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    ----------
+    udf_service :  UserDefinedFunctionServicer
+      The interface for the User Defined Function (UDFunction)
+        which will be exposed over gRPC.
+
+
+    server_options :  list[tuple[str, int]]
+        the custom options for configuring the grpc server
```
Comment (ab93) [inline pynumaflow/function/multiproc_server.py:56]: Args don't match though

File: pynumaflow/function/multiproc_server.py [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+                udfunction_pb2.Datum(
+                    keys=list(msgt.keys),
+                    value=msgt.value,
+                    event_time=udfunction_pb2.EventTime(event_time=event_time_timestamp),
+                    watermark=udfunction_pb2.Watermark(watermark=watermark_timestamp),
+                )
+            )
+        return udfunction_pb2.DatumList(elements=datums)
+
+    # def ReduceFn(
```
Comment (ab93) [inline pynumaflow/function/multiproc_server.py:165]: do we need to keep this?

File: pynumaflow/tests/function/test_async_server_err.py
Comment (ab93) [inline pynumaflow/tests/function/test_async_server_err.py:1]: We can move this to the `test_async_server.py` file, right?

File: pyproject.toml [hunk 6]
```diff
@@ -25,6 +25,8 @@ grpcio-tools = "^1.48.1"
 google-cloud = "^0.34.0"
 google-api-core = "^2.11.0"
 protobuf = ">=3.20,<5.0"
+attr = "^0.3.2"
```
Comment (ab93) [inline pyproject.toml:28]: where are we using this?

**Review by ab93 — COMMENTED** [review]
Some clarifying comments :)

**Comment by vigith** [comment]
close in favor of #81

**Pull request closed without merging**